ts/nni_manager/training_service/reusable/dlc/dlcClient.ts

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+'use strict';
+
+import { Deferred } from 'ts-deferred';
+import { PythonShell } from 'python-shell';
+import { getLogger, Logger } from '../../../common/log';
+
+export class DlcClient {
+    private log: Logger;
+    public type: string;
+    public image: string;
+    public podCount: number;
+    public ecsSpec: string;
+    public region: string;
+    // e.g., data1e6vg1tu0zi7, to generate it, go to 'Dataset Config' page of DLC
+    //       create a NAS data and copy the 'DataSet ConfigurationID'
+    public nasDataSourceId: string;
+    public accessKeyId: string;
+    public accessKeySecret: string;
+    public experimentId: string;
+    public environmentId: string;
+    public userCommand: string;
+    public pythonShellClient: undefined | PythonShell;
+
+    constructor(
+        type: string,
+        image: string,
+        podCount: number,
+        experimentId: string,
+        environmentId: string,
+        ecsSpec: string,
+        region: string,
+        nasDataSourceId: string,
+        accessKeyId: string,
+        accessKeySecret: string,
+        userCommand: string,
+        ) {
+        this.log = getLogger('DlcClient');
+        this.type = type;
+        this.image = image;
+        this.podCount = podCount;
+        this.ecsSpec = ecsSpec;
+        this.image = image;
+        this.region = region;
+        this.nasDataSourceId = nasDataSourceId;
+        this.accessKeyId = accessKeyId;
+        this.accessKeySecret = accessKeySecret
+        this.experimentId = experimentId;
+        this.environmentId = environmentId;
+        this.userCommand = userCommand;
+    }
+
+    private getScript(): string[] {
+        const script: string[] = [];
+        script.push(
+            `python ./config/dlc/dlcUtil.py --type ${this.type} --image ${this.image} --pod_count ${this.podCount} ` +
+            `--ecs_spec ${this.ecsSpec} --experiment_name nni_exp_${this.experimentId} ` +
+            `--region ${this.region} --nas_data_source_id ${this.nasDataSourceId} --access_key_id ${this.accessKeyId} ` + 
+            `--access_key_secret ${this.accessKeySecret} --user_command "${this.userCommand}"` );
+        return script;
+    }
+
+    public submit(): Promise<string> {
+        const deferred: Deferred<string> = new Deferred<string>();
+        this.pythonShellClient = new PythonShell('dlcUtil.py', {
+            scriptPath: './config/dlc',
+            pythonPath: 'python',
```

Comment:
Detect system OS, if it is windows use `python`, otherwise use `python3`.

Reply:
in dlc mode, only Linux environment is available, no windows case.